src/coprocessor/dag/executor/aggregate.rs

```diff
@@ -423,4 +463,28 @@ mod tests {
         }
         assert_eq!(aggr.c, 3);
     }
+
+    #[test]
+    fn test_variance() {
+        let mut var = Variance {
+            sum: 0.0,
+            variance: 0.0,
+            cnt: 0,
+        };
+        let cfg = EvalConfig::default_for_test();
+        let mut ctx = EvalContext::new(Arc::new(cfg));
+        let data = vec![
+            Datum::U64(1),
+            Datum::I64(2),
+            Datum::F64(12.1),
+            Datum::Null,
+            Datum::Dec(Decimal::from_f64(1.234).unwrap()),
+        ];
+        for v in data {
+            var.update(&mut ctx, &mut vec![v]).unwrap();
+        }
+        assert_eq!(var.cnt, 4);
+        assert_eq!(var.sum, 16.334);
+        assert_eq!(var.variance, 86.232867);
```

Comment:
consider comparing them within some error: `(* left_val - * right_val).abs() < error ` or convert them to `Datum::F64`